runtime/compiler/control/CompilationThread.cpp

```diff
@@ -3802,6 +3834,11 @@ TR::CompilationInfoPerThread::processEntries()
             if (compInfo->getPersistentInfo()->getJITaaSMode() == CLIENT_MODE && !client)
                {
                client = new (PERSISTENT_NEW) JITaaS::J9ClientStream(compInfo->getPersistentInfo());
+               if (!compInfo->checkVersion(client))
+                  {
+                     if (TR::Options::getVerboseOption(TR_VerboseJITaaS))
+                         TR_VerboseLog::writeLineLocked(TR_Vlog_JITaaS, "version check failed \n");
+                  }
```

Comment:
What happens now that the version check failed? We need to perform the compilations locally.